### ISSUE TITLE
redis-connection compatible with bluebird promisifyall ? 

### ISSUE THREAD

**Issue by sohilpandya** [dwyl/redis-connection#19]
@nelsonic just wanted to find out if we could use redis-connection and promisify everything.

**Comment by nelsonic** [comment]
@sohilpandya you *probably* could... why?

**Comment by sohilpandya** [comment]
We are struggling to use redis-connection with bluebird to promisify redis functions.

This is how we tried to use it, but getting an error in the console. hence we are having to use npm module `redis` to make it work, but trying to make our tests pass with this module is problematic. 
Here is our code where we are trying to make it work with redis-connection [L1-L6](https://github.com/Unitie-FAC/Unitie-MVP/blob/master/lib/societydashboard.js#L1-L6)
```
var redisClient = require('redis-connection')();
bluebird.promisifyAll(redis.RedisClient.prototype);
bluebird.promisifyAll(redis.Multi.prototype);
```

Would prefer to keep using `redis-connection`. hopefully there is a way to implement it. 

Thanks.

**Comment by nelsonic** [comment]
Right, but what is the use-case that requires you to "*promisify*" the Redis requests?

**Comment by sohilpandya** [comment]
without promisifing redis this is how our code would look with the use of `redis-connection` and callbacks (with better naming hopefully :))
```
          var array = [];
          var callback2 = function() {
            console.log(array);
            return reply(array);
          };
          var callback = function(replyarray) {
            var itemProcessed = 0;
            replyarray.forEach(function(elem) {
              redisClient.hgetall(elem, function(err, replyobj) {
                array.push(replyobj);
                itemProcessed++;
                if (itemProcessed === replyarray.length) {
                  callback2();
                }
              });
            });
          };
          redisClient.smembers('type:society', function(err, replyarray) {
            callback(replyarray);
          });
```

It is not that bad, could be made better, but still a couple of callbacks and hence we thought it might be a good time to look into bluebird with redis, we know our problem is not that complex, hence @thegsi and I thought it would ease us into using promises. This is how it looks with promises and `redis` module
```
          var object = [];
          var test = client.smembersAsync('type:society')
            .each(function(membersArray) {
              return client.hgetallAsync(membersArray)
                .then(function(hgetallobject) {
                  object.push(hgetallobject);
                });
            })
            .then(function() {
              // client.quit();
              return reply(object);
            });
```

a lot neater and easier to read (I think :worried: ), although testing is causing some issues, when we use `redis` testing locally is a bit of a mess due to open connections and hence we are having to quit client, something we don't have to do every time in `redis-connection`

In short we thought it would be a good time to look into it, not something that the client had asked for. will definitely look into ES6 promises as well. 

Thanks for that.

**Comment by nelsonic** [comment]
@sohilpandya how did you end up solving this issue?

**Comment by sohilpandya** [comment]
@nelsonic we created a single file that holds the server, then require the same server in all different places. :smile:

**Comment by nelsonic** [comment]
Cool. please paste link then close issue if you are happy with how it turned out. :+1:

**Comment by sohilpandya** [comment]
https://github.com/Unitie-FAC/Unitie-MVP/blob/master/lib/redis.js#L1-L13

Exporting the above file, solved the issue for us.
